src/backend/python_nlp/smart_filters.py

```
@@ -115,7 +115,9 @@ def __init__(self, db_path: str = None):
             db_path = os.path.join(DATA_DIR, filename)
 
         # Validate the final path
-        self.db_path = str(PathValidator.validate_database_path(db_path, DATA_DIR))
+        if db_path != ":memory:" and not PathValidator.is_safe_path(DATA_DIR, db_path):
+             raise ValueError(f"Invalid database path: {db_path}")
```

Flake8 (7.3.0)

[error] indentation is not a multiple of 4

(E111)